05week/spaceTravelToMars.js

```diff
@@ -9,7 +9,46 @@ let jobTypes = {
   programmer: 'Any Ship!'
 };
 
-// Your code here
+//create crewmember class 
+class CrewMember {
+  constructor(name, job, specialSkill){
+    this.name = name;
+    this.job = job ; //(jobTypes)
+    this.specialSkill = specialSkill;
+    this.ship = null;
+  } 
+  
+  enterShip(anyShip){
+    // add this.CrewMember 
+    //add entire variable ship & crew
+    this.ship = anyShip;
+    anyShip.crew.push(this);
+    console.log(anyShip)
+
+  }
+
+}
+
+class Ship {
+  constructor(name, type, ability){
+    this.name = name;
+    this.type = type;//(jobTypes)
+    this.ability = ability; 
+    this.crew = [];
+  }
+  
+  // this method returns ship's ability IF 
+  missionStatement(){
+    for(let i = 0; i < this.crew.length; i++){
+      console.log(jobTypes[this.crew[i].job])
+      if(this.crew.length > 0 && jobTypes[this.crew[i].job] == this.type){
```

Comment:
you do not need to check the crew.length in the if statement because you would not be inside the loop if it wasn't greater than zero